liblcm-test/meson.build

```
@@ -0,0 +1,41 @@
+lcm_sink_exe = executable('lcm-sink', 'lcm-sink.c',
+  dependencies : lcm_lib_dep,
+  install : true)
+
+lcm_source_exe = executable('lcm-source', 'lcm-source.c',
+  dependencies : lcm_lib_dep,
+  install : true)
+
+if unix
+  lcm_tester_exe = executable('lcm-tester', 'lcm-tester.c',
+    dependencies : [glib_dep, lcm_lib_dep],
+    install : true)
+endif
+if windows
+  lcm_tester_exe = executable('lcm-tester', 'lcm-tester.c',
+    dependencies : [glib_dep, wsock32, ws2_32, lcm_lib_dep],
+    install : true)
+endif
+
+if unix
+  lcm_example_exe = executable('lcm-example', 'lcm-example.c',
+    dependencies : lcm_lib_dep,
+    install : true)
+endif
+if windows
+  lcm_example_exe = executable('lcm-example', 'lcm-example.c',
+    dependencies : [ws2_32, lcm_lib_dep],
+    install : true)
+endif
```

It looks like this could be simplified a bit by making an extra deps variable:

```suggestion
extra_deps = []
if windows
  extra_deps = [glib_dep, lcm_lib_dep]
endif

lcm_tester_exe = executable('lcm-tester', 'lcm-tester.c',
  dependencies : [glib_dep, lcm_lib_dep, extra_deps],
  install : true)

lcm_example_exe = executable('lcm-example', 'lcm-example.c',
  dependencies : [lcm_lib_dep, extra_deps],
  install : true)
```

This type of pattern occurs elsewhere. Perhaps it could be applied there as well?

---

This doesn't shorten the code because the extra deps for Windows are wsock32 and ws2_32, but only one executable uses both.

---

`lcm-example` for Windows doesn't use wsock32, and I would rather not add dependencies that it doesn't need. Then, adding another extra_deps list to fix it would make the code longer.

---

I shortened the code in a different way, by consolidating the two if statements into one: 8d1c9346f846d8b8d46b1becf04875afc1ad24f7